components/elm/src/main/surfrdMod.F90

```diff
@@ -886,7 +873,22 @@ subroutine surfrd_special(begg, endg, ncid, ns,ntpu)
          end do
       
     end do
+    
+    ! Obtain firrig and surface/grnd irrigation fraction
+    if (firrig_data) then
```

Comment:
The "firrig_data" was a flag indicating whether the surface data includes firrig, f_surf, and f_grd information. If not, then the default values (0.7, 1,0) will be used, so the user can still run the irrigation scheme. I would suggest keeping [lines 176-178](https://github.com/E3SM-Project/E3SM/pull/5016/files#:~:text=domain%25firrig%20%20%20%3D,%25f_grd%20%20%20%20%3D%200.0_r8) with the updated tpu indexing.